### PR TITLE
Mobile LH doc upload: adjust tracked item id param

### PR DESCRIPTION
**Pull request by aherzberg**
## Summary
New LH service allows for multiple tracked item ids while EVSS did not. Mobile was sending singular version of the parmeter causing it to never pick up the parameter. 

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/7154

## Testing done
specs

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
mobile

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/mobile/spec/request/claims_document_upload_spec.rb
Comment (aherzberg) [inline modules/mobile/spec/request/claims_document_upload_spec.rb:1]: I was having a lot of trouble checking the Lighthouse Document object itself had the tracked item id through rspec using `expect_any_instance_of` so didn't include it but did check it manually.

File: modules/mobile/app/controllers/mobile/v0/claims_and_appeals_controller.rb [hunk 1]
```diff
@@ -82,6 +82,7 @@ def request_decision
       def upload_document
         jid = if Flipper.enabled?(:mobile_lighthouse_document_upload, @current_user)
                 params[:claim_id] = params[:id]
+                params[:tracked_item_ids] = Array.wrap(params[:tracked_item_id])
```
Comment (kpethtel) [inline modules/mobile/app/controllers/mobile/v0/claims_and_appeals_controller.rb:85]: One small thing I'll point out is that we're passing extra, unused fields to the upstream service. I'm sure it just ignores them, but I think alternatively we could probably do something like 
`params[:tracked_item_ids] = Array.wrap(params.delete(:tracked_item_id))`
which would prevent us from also sending `tracked_item_id`

**Review by kpethtel — DISMISSED** [review]
👍

File: modules/mobile/app/controllers/mobile/v0/claims_and_appeals_controller.rb (hunk 1 above)
Comment (aherzberg) [reply modules/mobile/app/controllers/mobile/v0/claims_and_appeals_controller.rb]: good point. updated to delete the parameter. Also found a way to test for the tracked item ids so added that.

File: modules/mobile/spec/request/claims_document_upload_spec.rb [hunk 2]
```diff
@@ -86,7 +88,7 @@
   end
 
   it 'normalizes requests with a null tracked_item_id' do
-    params = { file:, claim_id:, tracked_item_ids: 'null', documentType: document_type }
+    params = { file:, claim_id:, tracked_item_id: nil, documentType: document_type }
```
Comment (aherzberg) [inline modules/mobile/spec/request/claims_document_upload_spec.rb:91]: we had a test all along for tracked item id but it was only testing for nil which it always was but we never tested for a present one. the string 'null' doesn't seem like a valid case as I'm unsure if the id is an integer or not so am hesitant to try to force it into one. passing nil makes the parameter an empty string (which seems weird) but I added some safety guards against it so we don't pass it to the LH service

**Review by kpethtel — APPROVED** [review]

**Pull request merged**